src/sentry/monitors/migrations/0011_backfill_monitor_environment_is_muted.py

```diff
@@ -0,0 +1,60 @@
+# Generated by Django 5.2.1 on 2025-11-13 19:27
+
+from django.db import migrations
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def backfill_monitor_environment_is_muted(
+    apps: StateApps, schema_editor: BaseDatabaseSchemaEditor
+) -> None:
+    """
+    Backfill MonitorEnvironment.is_muted from Monitor.is_muted for all muted monitors.
+
+    After this migration:
+    - is_muted=True monitors will have ALL environments muted
+    - is_muted=False monitors will have environments unchanged (some, none, or all unmuted)
+
+    This gets us into the correct state for the dual-write implementation where:
+    - Monitor is muted if and only if ALL environments are muted
+    - Monitor is unmuted if ANY environment is unmuted
+    """
+    Monitor = apps.get_model("monitors", "Monitor")
+    MonitorEnvironment = apps.get_model("monitors", "MonitorEnvironment")
+
+    # Only need to update environments for muted monitors since unmuted is already the default
+    muted_monitors = Monitor.objects.filter(is_muted=True)
+
+    for monitor in RangeQuerySetWrapperWithProgressBar(muted_monitors):
```

Comment:
You're better off filtering `is_muted` in memory here instead of the query, otherwise you might end up with timeouts.